src/ERC3009.sol

```diff
@@ -48,18 +115,90 @@ abstract contract ERC3009 is IERC3009, StatefulERC712 {
         bytes32 r_,
         bytes32 s_
     ) external {
-        _transferWithAuthorization(
-            TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+        _transferWithAuthorizationBefore(from_, validAfter_, validBefore_, nonce_);
+
+        _revertIfInvalidSignature(
+            from_,
+            _getDigest(
+                keccak256(
+                    abi.encode(
+                        TRANSFER_WITH_AUTHORIZATION_TYPEHASH,
+                        from_,
+                        to_,
+                        value_,
+                        validAfter_,
+                        validBefore_,
+                        nonce_
+                    )
+                )
+            ),
+            abi.encodePacked(r_, s_, v_)
```

Comment:
I am confused here, too. Why do we need to `abi.encodePacked(r_, vs_)` at all? Is there no (and can there not be be) a `_revertIfInvalidSignature` that takes and uses `r`, `s`, and `v` without ever needing to encode?

Reply:
Fixed in this commit: https://github.com/MZero-Labs/common/pull/8/commits/b63a1861b99ff78cc8356262b8076badf40154a4